api/collections/views.py

```diff
@@ -346,7 +347,7 @@ class CollectedMetaDetail(JSONAPIBaseView, generics.RetrieveUpdateDestroyAPIView
     def get_object(self):
         cgm = get_object_or_error(
             CollectionSubmission,
-            self.kwargs['cgm_id'],
+            Q(collection=Collection.load(self.kwargs['collection_id']), guid___id=self.kwargs['cgm_id']),
```

Comment:
Benchmarks - I went with #3 - speeds of these are very similar, but also this allows us to use `get_object_or_error` without any other modifications.  `get_object_or_error` returns a 404 if the CollectionSubmission object is not found, but #1 for example, would throw a Guid error which we'd have to separately handle. 
```python
1. 0.036 ms - Q(collection_id=Guid.objects.get(_id=self.kwargs['collection_id']).referent.id, guid___id=self.kwargs['cgm_id'])

2. 0.023 ms - Q(collection__guids=Guid.load(self.kwargs['collection_id']), guid___id=self.kwargs['cgm_id'])

3. 0.019 ms - Q(collection=Collection.load(self.kwargs['collection_id']), guid___id=self.kwargs['cgm_id'])